### PR TITLE
Fixed NRE on singleplayer accepted bounties

### PR DESCRIPTION
**Pull request by awebbdev**
I've tested this several times now and haven't run into an issue so far.  While I didn't rollback the changes that caused the original issue, I did remove the IsDedicated check in the if statements since almost all the following behavior only check if it is a server.  Let me know if you have any questions about my code, this is my first time working with a Valheim mod.

### CODE REVIEW

**Review by RandyKnapp — APPROVED** [review]

File: EpicLoot/Adventure/BountyTarget.cs [hunk 1]
```diff
@@ -51,17 +51,25 @@ private bool HasBeenSetup()
...
-            _bountyInfo.ToPackage(pkg);
+            if (ZNet.instance.IsServer() || !ZNet.instance.IsServer() && !ZNet.instance.IsDedicated())
+            {
+                var pkg = new ZPackage();
+                _bountyInfo.ToPackage(pkg);
 
-            EpicLoot.LogWarning($"SENDING -> RPC_SlayBountyTarget: {_monsterID} ({(_isAdd ? "minion" : "target")})");
-            ZRoutedRpc.instance.InvokeRoutedRPC("SlayBountyTarget", pkg, _monsterID, _isAdd);
+                EpicLoot.LogWarning($"SENDING -> RPC_SlayBountyTarget: {_monsterID} ({(_isAdd ? "minion" : "target")})");
+                ZRoutedRpc.instance.InvokeRoutedRPC("SlayBountyTarget", _monsterID, _isAdd, pkg);
```
Comment (RandyKnapp) [inline EpicLoot/Adventure/BountyTarget.cs:60]: Bug I encountered in testing here, you switched the order of the parameters for this. Was there a reason why?

**Comment by RandyKnapp** [comment]
Did you test to see if this worked on a dedicated server?

File: EpicLoot/Adventure/BountyTarget.cs (hunk 1 above)
Comment (awebbdev) [reply EpicLoot/Adventure/BountyTarget.cs:60]: Sorry, was away for a bit.  I had originally tried a default null parameter for pkg so I put it last, I must have forgotten to put it back when I did my code clean-up.

**Comment by awebbdev** [comment]
I did test it on a dedicated server, just a quick dry run of a new server, went to merchant, the completed two bounties without issue.   I did not have an existing save where I could make sure existing  bounties were able to complete successfully.

**Comment by sbtoonz** [comment]
Second pass is a lot cleaner than my first pass

**Comment by Digitalroot** [comment]
Using the current version of the EL code base. I get the following when killing a bounty target. 

I get this error running the game SP, no server started or SP with server started from the client.

```
[Error  : Unity Log] EndOfStreamException: Unable to read beyond the end of the stream.
Stack trace:
System.IO.__Error.EndOfFile () (at <eae584ce26bc40229c1b1aa476bfa589>:0)
System.IO.BinaryReader.ReadByte () (at <eae584ce26bc40229c1b1aa476bfa589>:0)
System.IO.BinaryReader.Read7BitEncodedInt () (at <eae584ce26bc40229c1b1aa476bfa589>:0)
System.IO.BinaryReader.ReadString () (at <eae584ce26bc40229c1b1aa476bfa589>:0)
ZPackage.ReadString () (at <1b865f8b3455488592cdc401081112e3>:0)
ZRpc.Deserialize (System.Reflection.ParameterInfo[] paramInfo, ZPackage pkg, System.Collections.Generic.List`1[System.Object]& parameters) (at <1b865f8b3455488592cdc401081112e3>:0)
ZNetView.Deserialize (System.Int64 callerID, System.Reflection.ParameterInfo[] paramInfo, ZPackage pkg) (at <1b865f8b3455488592cdc401081112e3>:0)
RoutedMethod`3[T,U,V].Invoke (System.Int64 rpc, ZPackage pkg) (at <1b865f8b3455488592cdc401081112e3>:0)
ZRoutedRpc.HandleRoutedRPC (ZRoutedRpc+RoutedRPCData data) (at <1b865f8b3455488592cdc401081112e3>:0)
ZRoutedRpc.InvokeRoutedRPC (System.Int64 targetPeerID, ZDOID targetZDO, System.String methodName, System.Object[] parameters) (at <1b865f8b3455488592cdc401081112e3>:0)
ZRoutedRpc.InvokeRoutedRPC (System.Int64 targetPeerID, System.String methodName, System.Object[] parameters) (at <1b865f8b3455488592cdc401081112e3>:0)
ZRoutedRpc.InvokeRoutedRPC (System.String methodName, System.Object[] parameters) (at <1b865f8b3455488592cdc401081112e3>:0)
EpicLoot.Adventure.BountyTarget.OnDeath () (at <65777a7da40745f99767f4dec0233ae8>:0)
(wrapper dynamic-method) Character.DMD<Character::CheckDeath>(Character)
(wrapper dynamic-method) Character.DMD<Character::FixedUpdate>(Character)
Humanoid.FixedUpdate () (at <1b865f8b3455488592cdc401081112e3>:0)
```

Reverting to 0.8.0 returns to a working state.

Added to #284

**Comment by vitalikbyrevich** [comment]
Confirm. I also can't get a reward for completing a task. Only chests work. I had to abandon the tasks, but in the "active effects" menu, the tasks are still listed as unfulfilled, i.e. they were removed from the merchant, but they remained there.

**Pull request merged**